### ISSUE TITLE
Function type aliases not considered in `valid-title` and `prefer-describe-function-title`

### ISSUE THREAD

**Issue by FloEdelmann** [vitest-dev/eslint-plugin-vitest#692]
Consider this code:

```ts
// validator.ts
export type Validator = () => boolean
export const validator: Validator = () => true
```

```ts
// validator.spec.ts
import { validator } from './validator.ts'

describe('validator', () => { … })
```

With `typecheck: true`, no error is reported. However, I would have expected `prefer-describe-function-title` to suggest using `validator` instead of `'validator'` as the title. With `typecheck: false` (default), `prefer-describe-function-title` reports an error as expected.

In either case, when I change `'validator'` to `validator`, `valid-title` reports an error, telling me that `validator` is not a valid title, although it is a function and thus should be supported.

**Comment by JoshuaKGoldberg** [comment]
Amusing. I was hoping it would be more than a day after #690 that a bug report would be filed. 😄 

For context, #690 had `prefer-describe-function-title` reuse to the existing logic from `valid-title`:

https://github.com/vitest-dev/eslint-plugin-vitest/blob/7d9d12ce40f709c8503fa21efb3c37978666b675/src/utils/types.ts#L78-L91

This bug is fixed for me locally and lll existing tests pass if we switch to checking the type itself for call or construct signatures, rather than getting its symbol declaration:

```
export function isClassOrFunctionType(type: ts.Type): boolean {
  return (
    type.getCallSignatures().length > 0
    || type.getConstructSignatures().length > 0
  )
}
```
